### PR TITLE
Updates for building on Windows using Cygwin.

### PR DESCRIPTION
**Pull request by bill-chadwick**
Tested on Windows 7. ci target of make not yet working.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -329,3 +329,33 @@ Occasionally other changes to `master` might mean that your pull request cannot
...
+from make serve, but Chrome and Internet Explorer will.
+
+Microsoft Visual Studio's java script debugger may be used to debug the
+build/hosted/your-branch/examples. It will be convenient to set
+build/hosted/your-branch/examples/index.html as the startup page.
+
+Your ol3 source tree need not be under the Cygwin root.
+if you checkout to c:/ol3 then you can build under Cygwin at /cygdrive/c/ol3 .
+However, keep the path to the ol3 files short otherwise you may see
+ENAMETOOLONG errors.
```
Comment (elemoine) [inline CONTRIBUTING.md]: You could use back-ticks around `ENAMETOOLONG`.

File: Makefile [hunk 2]
```diff
@@ -23,7 +23,11 @@ CHECK_EXAMPLE_TIMESTAMPS = $(patsubst examples/%.html,build/timestamps/check-%-t
 
 TASKS_JS := $(shell find tasks -name '*.js')
 
-CLOSURE_LIB = $(shell node -e 'process.stdout.write(require("closure-util").getLibraryPath())')
+ifeq (CYGWIN,$(findstring CYGWIN,$(OS)))
+  CLOSURE_LIB := $(shell cygpath -u $(shell node -e 'process.stdout.write(require("closure-util").getLibraryPath())'))
```
Comment (elemoine) [inline Makefile]: I am not sure about the use of `:=`, as it will only work if the closure-util is already installed.

Comment (elemoine) [inline Makefile]: How about the following?

```
CLOSURE_LIB_NODE_CMD = "node -e 'process.stdout.write(require("closure-util").getLibraryPath())'"
ifeq (CYGWIN,$(findstring CYGWIN,$(OS)))
  CLOSURE_LIB = $(shell cygpath -u $(shell $(CLOSURE_LIB_NODE_CMD)))
else
  CLOSURE_LIB = $(shell $(CLOSURE_LIB_NODE_CMD))
endif

File: tasks/generate-info.js [hunk 3]
```diff
@@ -92,6 +99,11 @@ function getNewer(date, newer, callback) {
     callback(new Error('Trouble walking ' + sourceDir));
   });
   walker.on('end', function() {
+  
+    if (isWindows) {
+        paths = [sourceDir].concat(externsPaths);
+    }
+
```
Comment (elemoine) [inline tasks/generate-info.js]: Why is this needed?

File: Makefile (hunk 2 above)
Comment (bill-chadwick) [inline Makefile]: Those tweaks will have to wait till tonight if I am to make them. I did
mean  to use = not :=.
Bill

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -329,3 +329,33 @@ Occasionally other changes to `master` might mean that your pull request cannot
...
+
+Python is normally installed with Cygwin so need not be installed separately.
+By default Cygwin will use its own version of Python rather than Window's,
+so the Python modules should be installed for Cygwin's Python.
+
+The build targets `check-deps`, `serve`, `lint`, `build`, `test`, `check` and
+`host-examples` described above should all work. `host-examples` takes quite a
+while to run. If a target does not run properly first time, try it again.
+
+Currently, Firefox fails to run http://localhost:3000/build/examples ,
```
Comment (probins) [inline CONTRIBUTING.md]: don't think the comma at the end is needed

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -329,3 +329,33 @@ Occasionally other changes to `master` might mean that your pull request cannot
...
+from make serve, but Chrome and Internet Explorer will.
+
+Microsoft Visual Studio's java script debugger may be used to debug the
```
Comment (probins) [inline CONTRIBUTING.md]: javascript :-)

File: Makefile (hunk 2 above)
Comment (bill-chadwick) [inline Makefile]: elemoine,
Not sure the quotes are quite right in your
CLOSURE_LIB_NODE_CMD = "node -e 'process.stdout.write(require("closure-util").getLibraryPath())'"

It does not work in Cygwin

File: tasks/generate-info.js (hunk 3 above)
Comment (bill-chadwick) [inline tasks/generate-info.js]: It came from a previous windows build PR here 
https://github.com/openlayers/ol3/pull/2813/files#diff-47e08054189bdf70c82633c1a2479eca
looks like the comment block got lost. I am not sure all the changes in this module are needed anymore. I am running some tests.

File: Makefile (hunk 2 above)
Comment (elemoine) [inline Makefile]: Maybe you need to remove the double quotes entirely. I don't know.

File: tasks/generate-info.js (hunk 3 above)
Comment (bill-chadwick) [inline tasks/generate-info.js]: The change stops this error

node tasks/build.js config/ol.json build/ol.js
ERR! Trouble generating info: The command line is too long.

**Comment by elemoine** [comment]
I am happy with this patch if it works for people on Windows. Any objection from anyone?

**Comment by probins** [comment]
in my original PR https://github.com/openlayers/ol3/pull/2591/files I also changed the comment to the getNewer function

**Comment by probins** [comment]
@bill-chadwick #2593 also changed the java cmd so it would not exceed the cmd line length. Is this not necessary? IMO a flagfile would be better for everyone not just Windows users.

**Comment by bartvde** [comment]
no objections from me @elemoine
